lib/declare_schema/model/index_definition.rb

```diff
@@ -33,28 +33,30 @@ def initialize(model, fields, **options)
 
       class << self
         # extract IndexSpecs from an existing table
-        # always includes the PRIMARY KEY index
+        # always includes the PRIMARY KEY index for Model unless it is a HABTM Model.
         def for_model(model, old_table_name = nil)
           t = old_table_name || model.table_name
 
-          primary_key_columns = Array(model.connection.primary_key(t)).presence or
+          habtm_model = model.is_a?(DeclareSchema::Model::HabtmModelShim)
+
+          primary_key_columns = Array(model.connection.primary_key(t)).presence || habtm_model or
             raise "could not find primary key for table #{t} in #{model.connection.columns(t).inspect}"
 
           primary_key_found = false
           index_definitions = model.connection.indexes(t).map do |i|
             model.ignore_indexes.include?(i.name) and next
-            if i.name == PRIMARY_KEY_NAME
+            if i.name == PRIMARY_KEY_NAME && !habtm_model
```

Comment:
will the habtm_model ever return a primary key? if no I think we can remove this guard

Reply:
ack 👍